modular_skyrat/modules/implants/code/augments_head.dm

```diff
@@ -0,0 +1,188 @@
+		hacking for you. Due to their complexity, the system does not appear to work on cyborgs."
+	icon = 'modular_skyrat/modules/implants/icons/implants.dmi'
+	icon_state = "hackerman"
+	slot = ORGAN_SLOT_BRAIN_ANTISTUN
+	zone = BODY_ZONE_HEAD
+	implant_overlay = null
+	implant_color = null
+	actions_types = list(/datum/action/cooldown/spell/pointed/hackerman_deck)
+	w_class = WEIGHT_CLASS_SMALL
+
+/datum/action/cooldown/spell/pointed/hackerman_deck
+	name = "Activate Ranged Hacking"
+	desc = "Click on any machine, excepting cyborgs, to hack them. Has a short range, only two tiles."
+	active_msg = "You warm up your Binyat deck, there's an idle buzzing at the back of your mind as it awaits a target."
+	deactive_msg = "Your hacking deck makes an almost disappointed sounding buzz at the back of your mind as it powers down."
+	button_icon = 'modular_skyrat/modules/implants/icons/implants.dmi'
+	button_icon_state = "hackerman"
+	spell_requirements = SPELL_REQUIRES_MIND
+	cast_range = INFINITY // The range code doesn't work here, so we use our own, don't worry about it
+	aim_assist = FALSE
+	spell_max_level = 1 // God I hate actions
+	cooldown_time = 5 MINUTES
+	sparks_amt = 2
+	ranged_mousepointer = 'icons/effects/mouse_pointers/override_machine_target.dmi'
+	/// What we don't work on, will always not work on mobs because I know what you are
+	var/static/list/emag_blacklist = list(
+		/obj/machinery/shieldgen,
+		/obj/machinery/computer/communications,
+		/obj/machinery/computer/arcade,
+		/obj/machinery/computer/holodeck,
+		/obj/machinery/computer/emergency_shuttle,
+	)
+	/// How far away we can hack things
+	var/hack_range = 2
+
+/datum/action/cooldown/spell/pointed/hackerman_deck/is_valid_target(atom/cast_on)
+	. = ..()
+
+	if(ismob(cast_on) || is_type_in_list(cast_on, emag_blacklist))
+		owner.balloon_alert(owner, "security too strong")
+		return FALSE
+
+	if(get_dist(owner, cast_on) > hack_range)
+		owner.balloon_alert(owner, "too far away")
+		return FALSE
+
+	return TRUE
+
+/datum/action/cooldown/spell/pointed/hackerman_deck/cast(atom/cast_on)
+	. = ..()
+
+	unset_click_ability(owner)
+
+	playsound(owner, 'sound/effects/light_flicker.ogg', 50, TRUE)
+	var/beam = owner.Beam(cast_on, icon_state = "light_beam", time = 5 SECONDS)
+
+	owner.visible_message(span_bolddanger("[owner.name] makes an unusual buzzing sound as the air between them and [cast_on] crackles."), \
+			span_bolddanger("The air between you and [cast_on] begins to crackle audibly as the Binyat gets to work."))
+
+	if(!do_after(owner, 5 SECONDS, cast_on, IGNORE_SLOWDOWNS))
+		qdel(beam)
+		StartCooldown(1 SECONDS) // Resets the spell to working after a second, just so its not spammed
+		return
+
+	if(!cast_on.emag_act(owner))
+		owner.balloon_alert(owner, "can't hack this!")
+		StartCooldown(1 SECONDS) // Resets the spell to working after a second, just so its not spammed
+		return
+
+	playsound(cast_on, 'sound/machines/terminal_processing.ogg', 15, TRUE)
```

Comment:
I feel like there should probably still be some kind of way for this to be traceable back to someone in particular, _especially_ for admins, as otherwise it's going to be really annoying to track. I'd also appreciate if this was added to the combat logs.